tests/test_orient_symmetric.py

```diff
@@ -0,0 +1,82 @@
+from unittest import TestCase
+
+import numpy as np
+from numpy import linalg, random
+
+from aspire.abinitio import CLSymmetryC3C4
+from aspire.source import Simulation
+
+
+class OrientSymmTestCase(TestCase):
+    def setUp(self):
+        L = 32
+        symm = "C4"
+        n_ims = 32
+        src = Simulation(L=L, n=n_ims, symmetry_type=symm)
+        self.cl_class = CLSymmetryC3C4(src, n_symm=4, n_theta=360)
+        self.n_ims = n_ims
+
+    def tearDown(self):
+        pass
+
+    def testGlobalJSync(self):
+        n_ims = self.n_ims
+
+        # Build a set of outer products of random third rows.
+        vijs, viis = self.buildOuterProducts(n_ims)[:2]
+
+        # J-conjugate some of these outer products (every other element).
+        J = np.diag((1, 1, -1))
+        vijs_conj, viis_conj = vijs, viis
+        vijs_conj[::2] = J @ vijs_conj[::2] @ J
+        viis_conj[::2] = J @ viis_conj[::2] @ J
+
+        # Synchronize vijs_conj and viis_conj.
+        vijs_sync, viis_sync = self.cl_class.global_J_sync(vijs_conj, viis_conj)
+
+        # Check that synchronized outer products equal original
+        # up to J-conjugation of the entire set.
+        if vijs[0].all() == vijs_sync[0].all():
+            np.allclose(vijs, vijs_sync)
+            np.allclose(viis, viis_sync)
+        else:
+            np.allclose(vijs, J @ vijs_sync @ J)
+            np.allclose(viis, J @ viis_sync @ J)
+
```

Comment:
don't you need a `self.assertTrue()` for these?

Reply:
Absolutely. Thanks for catching that. Definitely uncovered some bugs with that change.